src/table/builder.rs

```diff
@@ -242,8 +240,8 @@ mod tests {
 
         let idx = table.inner.read_table_index().unwrap();
 
-        for i in 0..idx.offsets.len() {
-            assert_eq!(block_first_keys[i], idx.offsets[i].key);
+        for (i, k) in block_first_keys.iter().enumerate().take(idx.offsets.len()) {
```

Comment:
What if `block_first_keys.len() < idx.offsets.len()`?

Reply:
Will fix it!

Reply:
@BusyJay  I think it should always be equal. If we didn't get that right. Then the test should fail. Is it correct ?

Reply:
I have added real unit tests last week. You may have to rebase your work on latest master so as to see if this change really works.

Reply:
I prefer preserving how original code works, by iterating only on offsets.

Reply:
However, the clippy complain about the original code. I'm checking it right now.

Reply:
> I think it should always be equal. If we didn't get that right. Then the test should fail. Is it correct ?

Yes. However after changing to iterator, even they are not equal, the test will not fail. You can either add an explicit assert on length or just dismiss the lint.

Reply:
I prefer to add the assert of length.

Reply:
If they have the same length, then `take(n)` is unnecessary, right? Perhaps you can write `block_first_keys..iter().zip(&idx.offsets)`.